### DISCUSSION TITLE
How to make view page in existing resource ?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [filamentphp/filament#13723] [General]
On laravel 10/ filament site I have a resource AdminActivityResource which was created 
without `--view` option

and in some moment I understood that there no need to edit it, just view so I try to make
it only with view page in file app/Filament/Resources/AdminActivityResource.php I remade :

```
    public static function getPages(): array
    {
        return [
            'index' => Pages\ListAdminActivities::route('/'),
            'create' => Pages\CreateAdminActivity::route('/create'),
            'view' => Pages\ViewAdminActivity::route('/{record}/view'),
//            'edit' => Pages\EditAdminActivity::route('/{record}/edit'),
        ];
    }

```
I manually I created file app/Filament/Resources/AdminActivityResource/Pages/ViewAdminActivity.php :

```
namespace App\Filament\Resources\AdminActivityResource\Pages;

use App\Filament\Resources\AdminActivityResource;
use Filament\Actions;
use Filament\Resources\Pages\ViewRecord;

class ViewAdminActivity extends ViewRecord
{
    protected static string $resource = AdminActivityResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Actions\EditAction::make(),
        ];
    }
}

```
But running resource I got error :

`    Route [filament.admin.resources.admin-activities.view] not defined.
`
1) In which way have I to add path manually/some console command?

2) I tried to remake edit page intp "view", but did not find how hide Save button... 
    
```
    "laravel/framework": "^10.48.12",
    "filament/filament": "^3.2-stable",

```
Thanks in advance!

**Comment by dissto** [comment]
You could just use the helper command and be done with it 🤓
[Adding a View page to an existing resource](https://filamentphp.com/docs/3.x/panels/resources/viewing-records#adding-a-view-page-to-an-existing-resource) 🤔

**Reply by PetroGromovo** [reply]
Yes, I used this command, but I still have the same error that route not found

**Reply by dissto** [reply]
Do have you cached routes, or maybe cached components? 🤔

**Reply by PetroGromovo** [reply]
I have  cleared cache if you mean that...

**Reply by leandrocfe** [reply]
if `php artisan filament:clear-cached-components` doesn't work, try to create another page and post the output

**Reply by PetroGromovo** [reply]
I run command :

`php artisan filament:clear-cached-components
`
Do you mean to create another Resource ? I tried and got the same error :

I upgrated my project to the latest version of packages:
```
       "filament/filament": "^v3.2.97",
        "laravel/framework": "^11.19",

```        
Is it error of some version? Have I restore to old versions ?

**Reply by PetroGromovo** [reply]
I found similar error on other page. Is it only me who has this error ?

**Reply by leandrocfe** [reply]
Could you try it in a fresh filament install?

**Reply by PetroGromovo** [reply]
Do you mean to create a new app with  filament installed with some resource and data ?

**Reply by leandrocfe** [reply]
yes

**Reply by chrimutz** [reply]
I ran into the same error, this helps:
`php artisan filament:clear-cached-components `

```
"filament/filament": "^3.2",
"laravel/framework": "^11.9",
```
